r/src/altrep.cpp

```diff
@@ -1098,9 +1098,9 @@ void test_arrow_altrep_set_string_elt(sexp x, int i, std::string value) {
 }
 
 // [[arrow::export]]
-logicals test_arrow_altrep_is_materialized(sexp x) {
+sexp test_arrow_altrep_is_materialized(sexp x) {
```

Comment:
For my own learning, would be interested to hear why changing this from `logicals` to `sexp` mattered, and same with the return value.

Reply:
Is there generally a heuristic you use for when to use `SEXP` versus the cpp11 version of the variable, and vice versa?